### PR TITLE
fix: Deprecation of package ioutil in Go 1.16

### PR DESCRIPTION
**Pull request by zaunist**
**What type of PR is this?**

/kind deprecation

**What this PR does / why we need it**:

Deprecation of package ioutil in Go 1.16

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/utils/roundtripper/roundtripper.go [hunk 1]
```diff
@@ -182,7 +181,7 @@ func (d *DefaultRoundTripper) CaptureRoundTrip(request Request) (*CapturedReques
 		fmt.Printf("Received Response:\n%s\n\n", formatDump(dump, "< "))
 	}
 
-	body, _ := iou.ReadAll(resp.Body)
+	body, _ := io.ReadAll(resp.Body)
```
Comment (Xunzhuo) [inline conformance/utils/roundtripper/roundtripper.go:184]: Side comment, why not check error here?

Comment (zaunist) [reply conformance/utils/roundtripper/roundtripper.go:184]: I have the same doubts. Let's see what other people think, should I add error checking?

**Comment by mmorel-35** [comment]
Hi !
You can also add this linter configuration in `.golangci.yml` to never miss that rule
```yaml
  gomodguard:
    blocked:
      modules:
        - io/ioutil:
            recommendations:
              - io
              - os
```

**Comment by zaunist** [comment]
> Hi ! You can also add this linter configuration in `.golangci.yml` to never miss that rule
> 
> ```yaml
>   gomodguard:
>     blocked:
>       modules:
>         - io/ioutil:
>             recommendations:
>               - io
>               - os
> ```

Thanks. I will do it later.

File: .golangci.yml [hunk 2]
```diff
@@ -36,7 +36,16 @@ linters-settings:
     check-shadowing: true
   misspell:
     locale: US
-    ignore-words:
+    ignore-words: []
+  gomodguard:
```
Comment (mmorel-35) [inline .golangci.yml:40]: You also need to add `gomodguard` between line 15 and 16

Comment (zaunist) [reply .golangci.yml:41]: Thanks, I have enabled gomodguard.

**Review by mmorel-35 — APPROVED** [review]

**Review by shaneutt — APPROVED** [review]
Thank you!

/lgtm

**Pull request merged**